### DISCUSSION TITLE
Synology DSM (7.2.1) and Authelia (4.38.2) throws a "not privilege" (sic) error

### DISCUSSION THREAD

**Discussion by DJBenson** [authelia/authelia#6902] [Troubleshooting] [answered]
I noticed by chance yesterday that my Authelia instance was down and on closer inspection this was due to a bug with the recent release. After I'd sorted out my config to fix the warnings and got Authelia back up and running, I tried out some apps and all worked with the exception of DSM which throws a rather ambiguous "not privilege" (sic) error after confirming the SSO login on the Authelia page.

I appreciate this is probably something on the DSM side but given it was working fine until the new Authelia release, and there hasn't been a DSM release recently, I wondered if anybody else had the same issue.

There are no logs in DSM to even show a login was attempted and I can log in via LDAP using the same credentials.

<img width="461" alt="image" src="https://github.com/authelia/authelia/assets/1013909/9fcc25a2-3073-4436-aa42-4aab8b858068">

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It sounds like the issue you're encountering with DSM following the Authelia update might be related to changes in the Authelia configuration or how Authelia interacts with DSM, especially considering that everything else is working as expected. Given that DSM hasn't had a recent release, the focus should indeed be on the changes introduced by the Authelia update.

First, ensure that your Authelia configuration aligns with the requirements for DSM integration, particularly regarding OpenID Connect and LDAP settings. Since DSM does not support automatically creating users via OpenID Connect, it's crucial that Authelia and DSM share the same LDAP server for user authentication, as mentioned in the provided context.

Here are a few steps and checks you can perform:

1. **Review Authelia's OpenID Connect Configuration**: Ensure that the OpenID Connect configuration for DSM in Authelia hasn't been altered in a way that would affect authentication. This includes verifying the `client_id`, `client_secret`, and redirect URIs. The `client_secret` should be securely stored and not exposed in plaintext, as recommended.

2. **Check LDAP Configuration**: Since you can log in via LDAP using the same credentials, it's unlikely the issue lies here, but it's worth double-checking that the Authelia and DSM configurations haven't been inadvertently changed during the update.

3. **Examine Authelia Logs**: While DSM might not show logs for the login attempt, Authelia's logs could provide insights into what's happening during the authentication process. Look for any errors or warnings that occur when you attempt to log in to DSM via SSO.

4. **Review Breaking Changes**: Carefully review the release notes of the Authelia version you've updated to for any breaking changes or migration steps that you might have missed. Sometimes, new versions introduce changes that require updates to the configuration files or adjustments in how Authelia is integrated with other services.

5. **Test with a Minimal Configuration**: If possible, test the DSM integration with a minimal Authelia configuration that's known to work (based on your previous setup before the update). This can help identify if the issue is with the configuration or elsewhere.

6. **Community Feedback**: Since you're curious if others have encountered the same issue, consider reaching out on Authelia's GitHub discussions or community forums. Other users might have faced similar problems and could offer solutions.

If after these checks and tests the issue persists, providing detailed logs from Authelia during a DSM login attempt could help further diagnose the problem.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
There are no reported breaking changes in the latest release, all reported issues so far have been fixed as they were bugs.

As far as the errors we'd need (debug) log messages from both sides to diagnose it more.

**Comment by DJBenson** [comment]
@james-d-elliott I was about to edit how I described this as I knew it would be picked up. What I had last night was a failure due to the webauthn bug which was promptly fixed, however I also tacked all the config changes at the same time, so indeed there were technically no breaking changes, just a bug which took my system down.

I've just spotted the following in the Authelia logs when I attempt login;

> Access to https://cloud.mydomain.uk/ (method GET) is not authorized to user anonymous, responding with status code 401" method=GET path=/api/verify remote_ip=xxx.xxx.xxx.xxx

> https://cloud.mydomain.uk/ requires 2FA, cannot be redirected yet" method=POST path=/api/firstfactor remote_ip=xxx.xxx.xxx.xxx

**Reply by james-d-elliott** [reply]
Ah makes sense. No worries at all. Just don't want actual bugs to go unreported (or misconceptions becoming an issue). You can technically ignore the deprecation for quite a while, they are just warnings and should continue to operate normally per our versioning policy with few noted exceptions. Still don't know how that bug wasn't caught, sorry about that.

What is `cloud.`?

**Reply by DJBenson** [reply]
Yeah I decided to sort out my config now rather than later as I can go months without having to touch Authelia as "it just works" so I leave it be, but then got caught by the webauthn bug - this is mostly just a backstory to "my system worked for months then broke" but the only bit now not working is the DSM integration.

cloud. is the domain name of my DSM box. It goes Cloudflare (DNS) -> NPM -> Authelia -> DSM.

Just to reiterate, everything else is working (and to quantify, that means Proxmox, Portainer, Wordpress and a handful of other OIDC clients as well as a tonne of other apps protected by Authelia).

**Reply by DJBenson** [reply]
OIDC config for this service;

> - client_id: synology-dsm
>   client_name: Synology DSM
>   client_secret: $argon2id$v=redacted
>   public: false
>   authorization_policy: two_factor
>   redirect_uris:
>     - https://cloud.mydomain.uk
>   scopes:
>     - openid
>     - profile
>     - groups
>     - email
>   userinfo_signed_response_alg: none

**Reply by james-d-elliott** [reply]
Very strange. Those log messages appear like you're proxying DSM and have it included in the redirect / forward auth flow. I have to sleep, but it seems strange that you'd include that if you're using OIDC.

**Reply by DJBenson** [reply]
This might be a quirk of how I have it set up then (or it might be wrong, but not to drive this point too much, it was working previously).

If I hit cloud.mydomain.uk I get redirected to auth.mydomain.uk and can login successfully via the Authelia login page. I then get redirected to the DSM login page where I can either log in with local/LDAP credentials or using OIDC. Here I select OIDC and I get the Authelia confirmation, and then it redirects to this error whereas it previously redirected to my session in DSM.

My NPM config is the standard affair from the docs;

> include /snippets/authelia-location.conf;
> 
> location / {
>     include /snippets/proxy.conf;
>     include /snippets/authelia-authrequest.conf;
>     proxy_pass $forward_scheme://$server:$port;
> }

I'm happy to be told it's been wrong all this time if I can learn how to do it correctly - the idea behind the current method is that I do not expose DSM's login system directly to the internet without going through Authelia first - I just don't trust it (DSM), whether that's right or rational is for another day 😉 

Enjoy the sleep, thanks for the help and kudos for the latest release, I was so happy to get credential management!

**Reply by DJBenson** [reply]
I temporarily removed the NPM config so DSM was exposed directly and it still throws the same error so possibly a red herring (all done in incognito mode to avoid cached results).

**Comment by james-d-elliott** [comment]
Can't recall if I asked you to try this adjustment to the server config. If you're not using caddy/traefik please let me know and give this a shot if you are.

```yaml
server:
  endpoints:
    authz:
      forward-auth:
        implementation: 'ForwardAuth'
        authn_strategies:
          - name: 'HeaderProxyAuthorization'
            schemes:
              - 'Basic'
          - name: 'CookieSession'
```

**Reply by DJBenson** [reply]
I'm using nginxproxymanager. There's a couple of comments in the thread where I post my config but also disabled it and it made no difference.

**Reply by james-d-elliott** [reply]
Are you using the new /api/authz endpoint?

**Reply by DJBenson** [reply]
Almost certainly not if it's a recent change. Let me RTFM and come back if I still have issues.

**Reply by james-d-elliott** [reply]
Instead of `/api/verify` I meant.

**Reply by DJBenson** [reply]
I'll be honest, I'm a bit lost now. I've put the endpoints in my Authelia config file and updated my npm snippet to use the /api/authz endpoint but I now get a 500 error.

**Comment by lbalogh** [comment]
I'm interested if you find a solution, because I have the same issue.

It _may_ have stopped working after a specific 4.38 beta update of Authelia, but as I recently did several changes to my environment (Authelia beta updates then 4.38 official update, DSM update, some DNS changes) it's nearly impossible for me to track when it stopped working. As DSM "remembered" me I didn't see the issue immediately.

Fact is, I completed Authelia 4.38 migration (I started from a fresh DB, all warnings in the config file corrected, adjusted the config file for authz / session multi domain, etc.)

**Reply by DJBenson** [reply]
If you open the logs on DSM and search for SSO you will see when you last successfully authenticated. In my case it was 03/03 so about two weeks ago so I'm 99% sure something in Authelia changed as I've not applied any updates to DSM in my environment.

Still looking into it.

**Reply by lbalogh** [reply]
Unfortunately my DSM logs were purged so I have no history. I'm also also looking into it and I'll update this post with my findings, if any

**Comment by DJBenson** [comment]
So I didn't have the endpoints in my Authelia configuration. I've added them as per the docs and, aside from getting a "dangerous site" alert from Chrome, the result is the same, the "not privilege" error. Proxmox and Portainer (both using OIDC) work still.

**Reply by james-d-elliott** [reply]
Does DSM open it in a whole page or a mini window?

**Reply by DJBenson** [reply]
A popup window.

**Reply by DJBenson** [reply]
<img width="452" alt="image" src="https://github.com/authelia/authelia/assets/1013909/ff27d068-8062-4ea2-888a-a5394225a2a4">

<img width="568" alt="image" src="https://github.com/authelia/authelia/assets/1013909/0214fc2c-649b-44f8-9646-be8a1b6e7a1c">

**Reply by james-d-elliott** [reply]
Aha, I think I know what this is. I plan to have a build to confirm.

**Reply by james-d-elliott** [reply]
Actually, you're on nginx and it'd help me. Can you configure the authelia portal with the following directives (and confirm those headers are no longer displayed on the authelia portal):

```
proxy_hide_header  Cross-Origin-Embedder-Policy;
proxy_hide_header  Cross-Origin-Opener-Policy;
proxy_hide_header  Cross-Origin-Resource-Policy;
```

**Reply by DJBenson** [reply]
They are indeed removed but no change in the outcome.

Before:
<img width="535" alt="image" src="https://github.com/authelia/authelia/assets/1013909/9515167b-4f30-4e7a-afa5-947dd2e87ba0">

After:
<img width="533" alt="image" src="https://github.com/authelia/authelia/assets/1013909/5e41d7ac-6181-4eef-81f9-0f4d133eeb4c">

**Reply by DJBenson** [reply]
Actually not quite true, I now get this;

<img width="346" alt="image" src="https://github.com/authelia/authelia/assets/1013909/18b2c754-c86e-454e-9564-e55d277336e9">

**Reply by DJBenson** [reply]
The DSM logs suggest a null username is being passed. Last successful login was 02/03 using SSO.

<img width="1217" alt="image" src="https://github.com/authelia/authelia/assets/1013909/ab1940ec-24e7-45fa-9f63-a12e736ce3c4">

Prior to the header change, the authentication request didn't seem to even be reaching DSM (or the SSO client) as there was nothing in the logs.

**Reply by james-d-elliott** [reply]
That's not an overly helpful log message because it's definitely not null if you request `profile`.

**Reply by lbalogh** [reply]
I'm not sure if your issue is the same as mine, but I got it (partly) working by removing 2 redirect URIs :

![image](https://github.com/authelia/authelia/assets/6897141/cfa816de-3222-4d95-91e1-3bf1846c5359)

At least it works now for my main DSM url.

In Authelia's client OIDC config, I also added `token_endpoint_auth_method: client_secret_post`, not sure if necessary but Authelia's log was showing an error without this parameter.

 Here is my config :

![image](https://github.com/authelia/authelia/assets/6897141/d67dbb71-034a-40cc-a7f8-fff248db4edc)

At the time of writing I'm on DSM 7.2.1-69057 Update 4 / Authelia 4.38.6

**Reply by DJBenson** [reply]
@lbalogh adding token_endpoint_auth_method to my config (no other changes) allowed me to log in!

I'm still getting the "Dangerous Site" prompt from Chrome though which I've reported to Google as a false positive.

<img width="497" alt="image" src="https://github.com/authelia/authelia/assets/1013909/8d20ea48-40e1-4842-9bdb-abf40dd22c78">

**Reply by lbalogh** [reply]
Great we're making progress ;)

@james-d-elliott should the `token_endpoint_auth_method: client_secret_post` option be added in the Synology integration documentation (https://www.authelia.com/integration/openid-connect/synology-dsm/) ?

Regarding the Dangerous site warning, I can't help you, but a quick Google search led to this : https://support.google.com/chrome/thread/14102223/my-website-got-dangerous-in-google-chrome?hl=en

**Reply by james-d-elliott** [reply]
If there is a logged error that indicates that is the fix then yes.

**Reply by lbalogh** [reply]
Sure here is the error I get without the `token_endpoint_auth_method: client_secret_post` option :

```
app-1    | time="2024-03-26T19:45:57+01:00" level=error msg="Access Request failed with error: Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). The request was determined to be using 'token_endpoint_auth_method' method 'client_secret_post', however the OAuth 2.0 client registration does not allow this method. The registered client with id 'synology-dsm' is configured to only support 'token_endpoint_auth_method' method 'client_secret_basic'. Either the Authorization Server client registration will need to have the 'token_endpoint_auth_method' updated to 'client_secret_post' or the Relying Party will need to be configured to use 'client_secret_basic'." method=POST path=/api/oidc/token remote_ip=192.x.x.x stack="github.com/authelia/authelia/v4/internal/handlers/handler_oidc_token.go:25                   OpenIDConnectTokenPOST\ngithub.com/authelia/authelia/v4/internal/middlewares/http_to_authelia_handler_adaptor.go:114 handleRouter.NewHTTPToAutheliaHandlerAdaptor.func25\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54                            handleRouter.(*BridgeBuilder).Build.func16.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:91                           SecurityHeadersNoStore.func1\ngithub.com/valyala/fasthttp@v1.52.0/userdata.go:57                                           (*userData).Get\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:30                           SecurityHeadersBase.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/cors.go:216                             handleRouter.(*CORSPolicy).Middleware.func26\ngithub.com/fasthttp/router@v1.5.0/router.go:441                                              (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14                       handleRouter.LogRequest.func40\ngithub.com/authelia/authelia/v4/internal/middlewares/errors.go:38                            RecoverPanic.func1\ngithub.com/valyala/fasthttp@v1.52.0/server.go:2374                                           (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:224                                        (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:196                                        (*workerPool).getCh.func1\nruntime/asm_amd64.s:1695                                                                     goexit"
```

**Reply by james-d-elliott** [reply]
Yeah it makes sense to add it, welcome to make a PR or I'll get to it at some stage too. Had planned to do some reorganization in that area later this week.

**Reply by lbalogh** [reply]
Thanks @james-d-elliott I _tried_ to make a PR

**Reply by DJBenson** [reply]
I've reported the issue to Google and it's already been resolved, I'm no longer getting the error.

Thanks for identifying the cause of the error logging in.

**Reply by mjc0608** [reply]
Hello, I still have this problem with `token_endpoint_auth_method` setting to `client_secret_post`. I notice the problem only in Safari, and it works fine in browsers with Chromium kernels (Chrome, Chromium, Edge).

I'm using Synology's web station to proxy the 9091 port of Authelia in the docker. Here's the log from docker:
```
2024/06/11 15:23:55 | stdout | time="2024-06-11T15:23:55-07:00" level=debug msg="Authorization Request with id '13db6c9e-6e8c-4133-befa-4d4c5fd5cd2c' on client with id 'synology-dsm' was successfully processed, proceeding to build Authorization Response" method=GET path=/api/oidc/authorization remote_ip=172.19.0.1
2024/06/11 15:23:55 | stdout | time="2024-06-11T15:23:55-07:00" level=debug msg="Authorization Request with id '13db6c9e-6e8c-4133-befa-4d4c5fd5cd2c' on client with id 'synology-dsm' is being processed" method=GET path=/api/oidc/authorization remote_ip=172.19.0.1
2024/06/11 15:23:54 | stdout | time="2024-06-11T15:23:54-07:00" level=debug msg="Successful 1FA authentication attempt made by user 'jcma'" method=POST path=/api/firstfactor remote_ip=172.19.0.1
2024/06/11 15:23:54 | stdout | time="2024-06-11T15:23:54-07:00" level=debug msg="Mark 1FA authentication attempt made by user 'jcma'" method=POST path=/api/firstfactor remote_ip=172.19.0.1
2024/06/11 15:23:27 | stdout | time="2024-06-11T15:23:27-07:00" level=debug msg="Authorization Request with id '360f37d0-2c19-403a-ac4a-cfd9117bb1a0' on client with id 'synology-dsm' is being processed" method=GET path=/api/oidc/authorization remote_ip=172.19.0.1
2024/06/11 15:21:14 | stdout | time="2024-06-11T15:21:14-07:00" level=info msg="Listening for non-TLS connections on '[::]:9091' path '/'" server=main service=server
```

It looks like the authentication is successful. However, this login request never reached synology, and I cannot find a log entry for that login.

Also, after the failure message, if I press Synology's login button again, it directly logs me in without popping the login window.

Using synology DSM 7.2.1-69057 Update 4 and Authelia 4.38.8.

**Reply by mjc0608** [reply]
Also noticed that Safari initially block the pop up window. I need to click allow pop up button to open that window.

**Reply by mjc0608** [reply]
When allowing pop-up for the synology address in Safari's setting. This issue went away. So it looks like there's some problem dealing with these reopened pop-ups.

**Reply by james-d-elliott** [reply]
Safari treats actions by the user as more trusted than a script, sounds like maybe Synology using a script to open popups between the auth code flow and them retrieving the token as evidenced by lack of token exchange in the above logs.

**Reply by BetyOops** [reply]
> I'm not sure if your issue is the same as mine, but I got it (partly) working by removing 2 redirect URIs :
> 
> ![image](https://private-user-images.githubusercontent.com/6897141/316960034-cfa816de-3222-4d95-91e1-3bf1846c5359.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4ODM1ODQsIm5iZiI6MTcyMDg4MzI4NCwicGF0aCI6Ii82ODk3MTQxLzMxNjk2MDAzNC1jZmE4MTZkZS0zMjIyLTRkOTUtOTFlMS0zYmYxODQ2YzUzNTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTNUMTUwODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDY0ZDNkYWRhZTRmMTZiYjYwZGRlNjM1ZDdkZTgyZGZhNWI4ZmExZjZkZjI0NjQxZTU1MWMzZjllYWRjOTBkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TPvBFH_RQmBAv_NnEcTdpA_DualX5SveX01uR8wlSw8)
> 
> At least it works now for my main DSM url.
> 
> In Authelia's client OIDC config, I also added `token_endpoint_auth_method: client_secret_post`, not sure if necessary but Authelia's log was showing an error without this parameter.
> 
> Here is my config :
> 
> ![image](https://private-user-images.githubusercontent.com/6897141/316961245-d67dbb71-034a-40cc-a7f8-fff248db4edc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4ODM1ODQsIm5iZiI6MTcyMDg4MzI4NCwicGF0aCI6Ii82ODk3MTQxLzMxNjk2MTI0NS1kNjdkYmI3MS0wMzRhLTQwY2MtYTdmOC1mZmYyNDhkYjRlZGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTNUMTUwODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzI0MzI4NWNiMzQzZjA0OWQxYzdmNTQ0NzY2YjQyYWU2MjVmNGE5OTRiOGU5OGE2YWVlYTBhYTJiM2NjNDNiYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2NIDtv78igzvIpROkGf5SzbXXeyh8WSrZX0RRf5Bztk)
> 
> At the time of writing I'm on DSM 7.2.1-69057 Update 4 / Authelia 4.38.6

Same problem here if i have more than 1 request_uris (dsm and photos for example).

I noticed when I want to connect with 2 request_uris for exemple on dsm.me.com, i have this "not privilege" popup but the popup's url is photos.me.com instead of dsm.me.com.

**Comment by myevit** [comment]
I have same issue with Azure

**Comment by lorenzschmid** [comment, marked as answer]
This seems to be an issue on Synology's side. Having the same issue as described here (adding Authelia as OpenID server for multiple Synology applications each with their own subdomain), I contacted their technical support and received the following answer (emphasis mine):

> Per our developers, our SSO server package does not currently support multiple Redirect URIs with third-party SSO integration like OpenID, so you would only be able to log in with the single Redirect URI.
> 
> As a result, i**f you already have listed multiple Redirect URIs, the one that works seems to always be the last one on the list,** which should explain why only the last one would work.
> 
> To work around this, you would need to log into DSM first, and then access the application and resources as needed.
> 
> Our developers have logged this as something to add functionality for in a future update, but no workaround is present at this time to get multiple redirects to work.

**Comment by PiAir** [comment]
Can confirm the problem existing on Synology DSM 7.2.2-72806 Update 3 and Authentic 2025.2.2
I had it (finally after upgrading from DSM 7.2.1 to 7.2.2) working for DSM, after I added Synology Drive it now only works for Drive (as the second Redirect URI) and no longer for DSM.

**Comment by bhorbowicz** [comment]
I contacted Synology support and successfully configured SSO on DSM with multiple Redirect URLs.
If you're using a reverse proxy, make sure internal communication uses **HTTPS**, not HTTP.
For DSM, use https://nas:5001 instead of http://nas:5000 - same applies to Photos and Drive.
If you're using a self-signed SSL certificate on Synology, remember to **disable certificate checking**.

Sample configuration for Traefik:

```yaml
http:
  serversTransports:
    insecure-transport:
        insecureSkipVerify: true
  services:
    dsm-service:
      loadBalancer:
        servers:
          - url: "https://192.168.0.1:5001"
        serversTransport: insecure-transport
  routers:
    dsm-router:
      entryPoints:
       - "websecure"
      rule: "Host(`dsm.yourdomain.com`)"
      service: dsm-service
      tls: true
```

**Reply by mrksr** [reply]
Adding to this, what finally made my configuration work was to follow [Authentik's advice](https://integrations.goauthentik.io/infrastructure/synology-dsm/#troubleshooting) and use Redirect URIs of the form `https://example.com` and NOT `https://example.com/#/signin`. Together with using https as suggested above, things finally worked.
